static/script.js

```diff
@@ -72,24 +74,37 @@ function enableAutoRefresh () { // eslint-disable-line no-unused-vars
 }
 
 function unfoldAll () { // eslint-disable-line no-unused-vars
-  window.setInterval(unfoldButton, 100)
   $('#unfold-all-li').removeClass('d-none')
-  let allUnfoldIds = $('#refs_list').attr('content').replace(/[\[\]'"\ ]/gm, '').split(',').sort().toString() // eslint-disable-line
+  let allIds = $('#refs_list').attr('content')                                  // eslint-disable-line
+                              .replace(/[\[\]'"\ ]/gm, '')                      // eslint-disable-line
+                              .split(',')                                       // eslint-disable-line
+                              .sort()                                           // eslint-disable-line
+                              .toString()                                       // eslint-disable-line
+  let idsToUnfold = allIds.split(',')
+  let newSessionStorageValue = ''
 
   $('#unfold-all').on('click', function (event) {
-    if (document.getElementById('unfold-all').checked === false) {
-      sessionStorage.setItem('unfoldedRows', '')
-      window.location.reload()
+    let currentUnfoldedRowsArray = sessionStorage.getItem('unfoldedRows').split(',')
+    if ($(this)[0].checked) {
+      newSessionStorageValue = allIds
+    } else {
+      newSessionStorageValue = ''
     }
-  })
-  function unfoldButton () {
-    let currentUnfoldedRows = sessionStorage.getItem('unfoldedRows').split(',').sort()
-    let currentUnfoldedRowsString = currentUnfoldedRows.toString()
-    if (allUnfoldIds !== currentUnfoldedRowsString) {
-      if ($('#unfold-all').is(':checked')) {
-        sessionStorage.setItem('unfoldedRows', allUnfoldIds)
-        window.location.reload()
-      }
+    let diffArray = idsToUnfold.filter(x => !currentUnfoldedRowsArray.includes(x))
+    if (diffArray.length === 0) {
+      diffArray = idsToUnfold
     }
-  }
+    for (let id of diffArray) {
```

Comment:
*for-of* loop is meant to be used with objects.
If you are dealing with array always use one of:

- `map`
- `filter`
- `reduce`
- `forEach`

In this case you would like to use `forEach` loop which iterates of every item in array and applies callback on it.
It can be rewritten to:

```javascript
diffArray.forEach(function(id) {
  $("#${id}").toggleClass('show').parent().toggleClass('active-border')

  let headingId = id.replace('collapse', 'heading')
  $("#${headingId}").toggleClass('unfold').toggleClass('collapsed')
})
```